app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,98 @@
+package org.mozilla.fenix.components
+
+import android.content.Context
+import android.content.Intent
+import android.content.IntentSender
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import androidx.test.core.app.ApplicationProvider
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.spyk
+import io.mockk.verify
+import kotlin.random.Random
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.mozilla.fenix.home.intent.StartSearchIntentProcessor
+
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PrivateShortcutCreateManagerTest {
+
+    @Test
+    fun `GIVEN shortcut pinning is supported WHEN createPrivateShortcut is called THEN create a pinned shortcut`() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+
+        val privateShortcutCreateManager = spyk(PrivateShortcutCreateManager)
+
+        val icon = privateShortcutCreateManager.getIcon(context)
+
+        `assert Icon is correctly built`(icon)
+
+        val intent = privateShortcutCreateManager.getHomeScreenActivityIntent(context)
+        `assert HomeActivityIntent correctly built`(intent)
+
+        val shortcut = privateShortcutCreateManager.getShortcut(context, icon, intent)
+        `assert shortcut is correctly built`(shortcut, icon, intent)
+
+        val intentSender:IntentSender = spyk(privateShortcutCreateManager.getIntentSender(context))
+
+        /**
+         *  Robolectric returns null for binder ue to which
+         *  intentSender.asBinder() calls in hashCode() fails with NullPointerException
+         */
+
+        val intentSenderHashCode = Random.nextInt()
+        every { intentSender.hashCode() } returns intentSenderHashCode
+
+        every { privateShortcutCreateManager.getIntentSender(context) } returns intentSender
+        every { privateShortcutCreateManager.getIcon(context) } returns icon
+        every { privateShortcutCreateManager.getHomeScreenActivityIntent(context) } returns intent
+        every { privateShortcutCreateManager.getShortcut(context, icon, intent) } returns shortcut
+
```

Comment:
IntentSender object created while running with Robolectric has a problem, it does not have the binder attached
Due to which it fails during verify {}

Also IntentSender does not provide anything to test against. So I decided to spy it and mocked the hashcode()

I am just returning objects created by SUT only (after asserting them that they are built correctly)

I mocked the result for intentSender.hashcode() because running with Robolectric will try to run the actual method , but the IntentSender object created by SUT under Robolectric does not have a binder attached due to which it fails with NullPointerException

this is the error

java.lang.NullPointerException
	at android.content.IntentSender.hashCode(IntentSender.java:295)
	at io.mockk.EqMatcher.hashCode(Matchers.kt)
	at java.util.AbstractList.hashCode(AbstractList.java:541)
	at io.mockk.InvocationMatcher.hashCode(API.kt:3754)
	at io.mockk.RecordedCall.hashCode(API.kt:3800)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.put(HashMap.java:612)

I think Robolectric returns null for binders
robolectric/robolectric#2579 (comment)